### ISSUE TITLE
Cilium policy inconsistently allows traffic between two services

### ISSUE THREAD

**Issue by adityabond** [cilium/cilium#13284]
## Bug report

**General Information**

- Cilium version (run `cilium version`)
Client: 1.8.2 aa42034f0 2020-07-23T15:02:39-07:00 go version go1.14.6 linux/amd64
Daemon: 1.8.2 aa42034f0 2020-07-23T15:02:39-07:00 go version go1.14.6 linux/amd64
- Kernel version (run `uname -a`)
 x86_64 x86_64 x86_64 GNU/Linux
- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...)
- Link to relevant artifacts (policies, deployments scripts, ...)
- Generate and upload a system zip:
```
curl -sLO https://git.io/cilium-sysdump-latest.zip && python cilium-sysdump-latest.zip
```

**How to reproduce the issue**

Hello everyone, I have cilium governing the traffic for my services.  I have a peculiar situation where traffic from one service to another is inconsistently disallowed. It would be great if anyone can help me understand why this is the case.

A cilium policy trace shows that the traffic is allowed:

```
root@ip-10-0-5-190:/home/cilium# cilium policy trace --src-endpoint 188 --dst-endpoint 1485
----------------------------------------------------------------
Tracing From: [k8s:app=baas-person, k8s:app.kubernetes.io/managed-by=spinnaker, k8s:app.kubernetes.io/name=baas, k8s:io.cilium.k8s.namespace.labels.field.cattle.io/projectId=p-hf4fj, k8s:io.cilium.k8s.policy.cluster=default, k8s:io.cilium.k8s.policy.serviceaccount=default, k8s:io.kubernetes.pod.namespace=baas] => To: [k8s:app=baas-auth, k8s:app.kubernetes.io/managed-by=spinnaker, k8s:app.kubernetes.io/name=baas, k8s:io.cilium.k8s.namespace.labels.field.cattle.io/projectId=p-hf4fj, k8s:io.cilium.k8s.policy.cluster=default, k8s:io.cilium.k8s.policy.serviceaccount=default, k8s:io.kubernetes.pod.namespace=baas] Ports: [0/ANY]

Resolving egress policy for [k8s:app=baas-person k8s:app.kubernetes.io/managed-by=spinnaker k8s:app.kubernetes.io/name=baas k8s:io.cilium.k8s.namespace.labels.field.cattle.io/projectId=p-hf4fj k8s:io.cilium.k8s.policy.cluster=default k8s:io.cilium.k8s.policy.serviceaccount=default k8s:io.kubernetes.pod.namespace=baas]
* Rule {"matchLabels":{"any:app":"baas-person","k8s:io.kubernetes.pod.namespace":"baas"}}: selected
    Allows to labels {"matchLabels":{"cidr:10.0.0.0/8":""}}
      No label match for [k8s:app=baas-auth k8s:app.kubernetes.io/managed-by=spinnaker k8s:app.kubernetes.io/name=baas k8s:io.cilium.k8s.namespace.labels.field.cattle.io/projectId=p-hf4fj k8s:io.cilium.k8s.policy.cluster=default k8s:io.cilium.k8s.policy.serviceaccount=default k8s:io.kubernetes.pod.namespace=baas]
    Allows to labels {"matchLabels":{"reserved:none":""}}
      No label match for [k8s:app=baas-auth k8s:app.kubernetes.io/managed-by=spinnaker k8s:app.kubernetes.io/name=baas k8s:io.cilium.k8s.namespace.labels.field.cattle.io/projectId=p-hf4fj k8s:io.cilium.k8s.policy.cluster=default k8s:io.cilium.k8s.policy.serviceaccount=default k8s:io.kubernetes.pod.namespace=baas]
      Allows port [{53 ANY}]
    Allows to labels {"matchLabels":{"any:app":"baas-auth","k8s:io.kubernetes.pod.namespace":"baas"}}
    Allows to labels {"matchLabels":{"any:app":"baas-person","k8s:io.kubernetes.pod.namespace":"baas"}}
    Allows to labels {"matchLabels":{"any:app":"baas-webhooks","k8s:io.kubernetes.pod.namespace":"baas"}}
    Allows to labels {"matchLabels":{"any:app":"baas_message_queue","k8s:io.kubernetes.pod.namespace":"baas"}}
    Allows to labels {"matchLabels":{"any:app.kubernetes.io/name":"rabbitmq","k8s:io.kubernetes.pod.namespace":"baas"}}
    Allows to labels {"matchLabels":{"any:app":"baas-ledger-builder","k8s:io.kubernetes.pod.namespace":"baas"}}
    Allows to labels {"matchLabels":{"any:app":"baas-ledger-api","k8s:io.kubernetes.pod.namespace":"baas"}}
      Found all required labels
1/26 rules selected
Found allow rule
Egress verdict: allowed

Final verdict: ALLOWED
```

It fails occasionally and this is the dropped flow details from Hubble:

```

Last Seen
11:31:45 -07:00 9/24/20
Forwarding Status
dropped
Direction
request
Source Pod name
baas-person-6747d644fb-hpk6p
Source Endpoint Labels
app.kubernetes.io/managed-by=spinnaker
app.kubernetes.io/name=baas
app=baas-person
io.cilium.k8s.namespace.labels.field.cattle.io/projectid=p-hf4fj
io.cilium.k8s.policy.cluster=default
io.cilium.k8s.policy.serviceaccount=default
namespace=baas
Source IP Address
10.0.4.141
Destination Pod name
baas-auth-57b8f77b75-gcd9h
Destination Endpoint Labels
app.kubernetes.io/managed-by=spinnaker
app.kubernetes.io/name=baas
app=baas-auth
io.cilium.k8s.namespace.labels.field.cattle.io/projectid=p-hf4fj
io.cilium.k8s.policy.cluster=default
io.cilium.k8s.policy.serviceaccount=default
namespace=baas
Destination IP Address
10.0.4.83
Destination Protocol/Port
TCP:8000
```

**Comment by aanm** [comment]
it's possible this will be fixed by https://github.com/cilium/cilium/pull/13263

**Comment by errordeveloper** [comment]
@adityabond have you tried to reproduce this with 1.8.3, 1.8.4 or 1.8.5?

**Comment by adityabond** [comment]
this is fixed by #13263

**Comment by adityabond** [comment]
> @adityabond have you tried to reproduce this with 1.8.3, 1.8.4 or 1.8.5?

Hi @errordeveloper,  this was fixed by #13263.
